src/node.ts

```diff
@@ -22,15 +16,15 @@ export class BPTreeNode {
 		return this.pointers.length === this.keys.length;
 	}
 
-	async readFrom(): Promise<number> {
-		let totalBytesRead = 0;
-
+	static async fromMemoryPointer(
+		mp: MemoryPointer,
+		resolver: RangeResolver
+	): Promise<{ node: BPTreeNode | null; bytesRead: number }> {
 		try {
-			console.log("Fetching initial data...");
-
-			let { data: sizeData } = await this.dataHandler({
-				start: 0,
-				end: 4,
+			let node = new BPTreeNode([], []);
```

Comment:
```suggestion
			const node = new BPTreeNode([], []);
```

For the most part we should try to prefer `const` whenever possible in js